packages/circuit-ui/components/Checkbox/Checkbox.tsx

```diff
@@ -175,20 +175,24 @@ export const Checkbox = forwardRef(
       onChange,
       children,
       value,
-      id: customId,
+      'id': customId,
       name,
       disabled,
       validationHint,
       className,
       style,
       invalid,
       tracking,
+      'aria-describedby': descriptionId,
       ...props
     }: CheckboxProps,
     ref: CheckboxProps['ref'],
   ) => {
     const id = customId || uniqueId('checkbox_');
     const validationHintId = uniqueId('validation_hint-');
+    const descriptionIds = `${
+      descriptionId ? `${descriptionId} ` : ''
+    }${validationHintId}`;
```

Comment:
Couldn't think of anything cleaner here. But in the end I think it's pretty clear, and tests also document the reasoning behind this. Feedback welcome.